starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,215 @@
+// You might reasonably want to keep some of these
+#include "macros.c"
+
+
+// Now for our data structures. These are intentionally left rather basic, so add whatever you want to them
+struct ship{
+	unsigned char x, y;
+	unsigned short halite;
+} ships[4096];
+
+struct player{
+	unsigned char dropoffs[64], dropoff_count, ship_count, ship_yard[2];
+	unsigned short ships[1024];
+	unsigned int halite;
+} players[4] = {0};
+
+struct map{
+	unsigned short halite;
+	unsigned short occupied; //(unsigned short)-1 means it's not occupied
+} map[64][64];
+
+
+// Now for some global variables to help us keep track of things
+unsigned char dropoffs[256][2], mhh, mhw, my_id, map_size[2], number_of_players;
+unsigned short max_turn, turn_number;
+
+FILE *log_file;
+
+
+// Now that all of the variables have been declared, we can include our default logic C file
+#include "default_logic.c"
+// This file can safely be removed or replaced with your own code
+
+
+int main(int argc, char* argv[]){
+	char *c;
+	unsigned char i, id, j, sca, scb, scc;
+	unsigned short s, ssa, ssb;
+	unsigned int sia;
+
+	// Firstly, we open a log file. If no log file is specified, it defaults to "MyBot.log"
+	if(argc != 2)
+		c = "MyBot.log";
+	else
+		c = argv[1];
+
+	log_file = fopen(c, "w");
+
+	// The first input we get from the engine is some JSON-encoded constants. This line can safely be skipped since the constants are, well, constants as far as the competition goes
+	if(scanf("%*s") != 0)
+		exit(1);
+
+	// Get number of players and player ID
+	if(scanf("%hhu %hhu", &number_of_players, &my_id) < 1)
+		exit(1);
+
+
+	// Now that we have our player ID, we add it to time(NULL) to seed our "smart" RNG
+	srand(time(NULL) + my_id);
+
+
+	// Get each player's ship yard point coordinates
+	for(i=0; i<number_of_players; i++){
+		if(scanf("%hhu %hhu %hhu", &sca, &scb, &scc) < 3)
+			exit(1);
+
+		players[sca].ship_yard[0] = scb;
+		players[sca].ship_yard[1] = scc;
+	}
+
+	// Read the map size
+	if(scanf("%hhu %hhu", &map_size[0], &map_size[1]) < 2)
+		exit(1);
+
+	// Now that we have the map size, let's set the map half width and map half height variables
+	mhw = map_size[0] >> 1;
+	mhh = map_size[1] >> 1;
+
+	// Figure out the max turn (We could just parse this from the JSON at the start, but that would be too easy)
+	switch(map_size[0]){
+		case 32:
+			max_turn = 400;
+		break;
+		case 40:
+			max_turn = 425;
+		break;
+		case 48:
+			max_turn = 450;
+		break;
+		case 56:
+			max_turn = 475;
+		break;
+		case 64:
+			max_turn = 500;
+		break;
+	}
+
+	// Finally, we read the initial map
+	for(i=0; i<map_size[0]; i++)
+		for(j=0; j<map_size[1]; j++)
+			if(scanf("%hu", &map[j][i].halite) < 1)
+				exit(1);
+
+	// Now that we've read the starting data, it's time to output our bot's name
+	printf("MyBot\n");
+	// stdout has to be flushed before the game engine will actually get the commands
+	fflush(stdout);
+
+
+	// Now we start the game loop
+	for(;;){
```

Comment:
I prefer `while(true)` because it's more readable although they compile to the same thing.

Reply:
C doesn't have boolean true/false without including a header file. Will while(1) work?